### PR TITLE
Add real world test suite

### PR DESCRIPTION
**Pull request by valeriupredoi**
Hey guys @bnlawrence @davidhassell here's a pretty serious test module that looks at:

- actual complex data with variable groups in one file
- a netCDF3 (or CLASSIC) file that **can not** be opened by `h5py`
- a native model dataset, that has `_FillValue` and `missing_value`
- I have also added a repo where we have test data `tests/test_data`

`Active()` does silly well in all these cases, it's brutal how good it performs! :beer:

### CODE REVIEW

**Comment by bnlawrence** [comment]
In the process of reviewing this, I realised we needed some really clean test data files for us and those responsible for the storage systems themselves, hence the last commit. Will review the rest of it properly tomorrow.

**Comment by bnlawrence** [comment]
@valeriupredoi Why do we get warnings from the `test_native_email_fails` test? E.g. 
```
tests/test_bigger_data.py::test_native_emac_model_fails
  /Users/BNL28/miniconda3/envs/pymb22a/lib/python3.9/site-packages/numpy/ma/core.py:5288: RuntimeWarning: Mean of empty slice.
    result = super().mean(axis=axis, dtype=dtype, **kwargs)[()]

tests/test_bigger_data.py::test_native_emac_model_fails
  /Users/BNL28/miniconda3/envs/pymb22a/lib/python3.9/site-packages/numpy/core/_methods.py:182: RuntimeWarning: invalid value encountered in divide
    ret = um.true_divide(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```
Surely the test should fail when it can't open the file, before we get to that point? Or is the data empty anyway and it's failing first?

**Comment by valeriupredoi** [comment]
@bnlawrence those are warnings that don't trigger a test fail unless I setup pytest to fail on warning; I don't think there is an actual issue coming from them to be honest, I think it's just that the data is fully masked in that particular slice - note that those warnings come from the pure-Numpy block

```python
    ncfile = str(test_data_path / "emac.nc")
    active = Active(ncfile, "aps_ave")
    active._version = 0
    d = active[4:5, 1:2]
    mean_result = np.mean(d)
```

that we can simply remove since the scope of the test is to show the fail on a Classic or netCDF3 file - what you reckon?

**Comment by valeriupredoi** [comment]
> In the process of reviewing this, I realised we needed some really clean test data files for us and those responsible for the storage systems themselves, hence the last commit. Will review the rest of it properly tomorrow.

Very good point, cheers!

**Comment by bnlawrence** [comment]
I figured the easiest way to handle this was in the test, rather than remove, although as you say the scope of the test doesn't cover it., but there is a bigger problem to be handled in active itself, which needs to handle this without errors.  I've dealt with it in [ee26bd](https://github.com/valeriupredoi/PyActiveStorage/pull/27/commits/eeb26bd85ffe1f4a2e25f91dc9f5c55a2cd58d69)

> @bnlawrence those are warnings that don't trigger a test fail unless I setup pytest to fail on warning; I don't think there is an actual issue coming from them to be honest, I think it's just that the data is fully masked in that particular slice - note that those warnings come from the pure-Numpy block
> 
> ```python
>     ncfile = str(test_data_path / "emac.nc")
>     active = Active(ncfile, "aps_ave")
>     active._version = 0
>     d = active[4:5, 1:2]
>     mean_result = np.mean(d)
> ```
> 
> that we can simply remove since the scope of the test is to show the fail on a Classic or netCDF3 file - what you reckon?

**Review by bnlawrence — APPROVED** [review]
I'm happy that this can be merged now, the outstanding bits are in other tickets.

File: tests/test_bigger_data.py [hunk 1]
```diff
@@ -157,7 +157,13 @@ def test_native_emac_model_fails(test_data_path):
...
     active._version = 0
     d = active[4:5, 1:2]
-    mean_result = np.mean(d)
+    if len(d):
+        mean_result = np.mean(d)
+    else:
+        # as it happens it is is possible for a slice to be
+        # all missing, so for the purpose of this test we 
+        # ignore it, but the general case should not.
+        pass
```
Comment (valeriupredoi) [inline tests/test_bigger_data.py:166]: ouch! Haha, this is a proper hack, but heyho - this bit of the test is not important at all, I'll have a proper fix in the future, cheers for it for now @bnlawrence

**Pull request merged**